### PR TITLE
[ACTIONS] Sapling AI new actions

### PR DESCRIPTION
**Pull request by jcortes**
### <samp>🤖 Generated by Copilot at 1feb632</samp>

This pull request adds a new integration for Sapling.ai, a natural language generation and editing platform. It introduces four action components for requesting and accepting completions and edits, and providing feedback to the Sapling system. It also adds a common app component and a constants module for handling the authentication and the API requests. It removes an unused component and updates the pnpm-lock.yaml file accordingly.

### CODE REVIEW

**Review by lcaresia — APPROVED** [review]
LGTM!

**Review by andrewjschuang — COMMENTED** [review]
`request-completion` action has failed:

Even though the action result is successful, there is no response returned. Maybe that could be a problem with the API?

image

**Review by andrewjschuang — COMMENTED** [review]
`accept-completion` action couldn't be tested since `request-completion` action didn't return a `completionId`

**Review by andrewjschuang — COMMENTED** [review]
`request-edits` action passed:

image
image

**Review by andrewjschuang — COMMENTED** [review]
`accept-edit` action passed:

image

**Review by andrewjschuang — COMMENTED** [review]
`reject-edit` action passed:

image

**Review by andrewjschuang — CHANGES_REQUESTED** [review]
Hi @jcortes, some tests have failed or need improvements!
Please check out the comments above

File: components/sapling_ai/package.json [hunk 1]
```diff
@@ -1,17 +1,19 @@
 {
   "name": "@pipedream/sapling_ai",
-  "version": "0.0.1",
+  "version": "0.0.2",
```
Comment (GTFalcao) [inline components/sapling_ai/package.json:3]:
```suggestion
  "version": "0.1.0",
```
As per the versioning guidelines, new (backwards-compatible) features should warrant a new `minor` version (not a `patch` version)

**Review by GTFalcao — COMMENTED** [review]
Hi @jcortes , thanks for your contribution! Looks good to me!

I've commented on the package.json version so we follow [our guidelines here](https://pipedream.com/docs/components/guidelines/#versioning), but I'm moving this forward since that is unrelated to the code and functionality.

**Review by andrewjschuang — COMMENTED** [review]
`accept-completion` action passed:

image
image

**Review by andrewjschuang — DISMISSED** [review]
Hello @jcortes, all tests have passed!

I have some improvement suggestions to make though:

- In `request-completion`, export `sessionId`
- In `accept-completion`, make `sessionId` optional too, as it seems to not matter that much. Also export it

**Review by andrewjschuang — APPROVED** [review]
Perfect, thank you very much!

**Pull request merged**